File: cashflow/src/main/java/com/backend/cashflow/util/verification/EmailVerify.java
package com.backend.cashflow.util.verification;

import org.springframework.stereotype.Component;

@Component
public class EmailVerify {

    /*
     * Verificador formato de E-mail;
     * @param String
     * @return boolean
     * */
    public boolean isEmailValid(String email) {
        String emailPattern = "[a-zA-Z0-9._-]+@[a-z]+\\.+[a-z]+";
        return email.matches(emailPattern);
    }
}

File: cashflow/src/main/java/com/backend/cashflow/verification/EmailVerify.kt
package com.backend.cashflow.verification;

import org.springframework.stereotype.Component;

@Component
class EmailVerify {
    /*
     * Verificador formato de E-mail;
     * @param String
     * @return boolean
     * */
    fun isEmailValid(email: String): Boolean {
        val emailPattern: String = "[a-zA-Z0-9._-]+@[a-z]+\\.+[a-z]+";
        return email.matches(emailPattern.toRegex());
    }
}
